elixir-mode.el

```diff
@@ -420,7 +420,8 @@ is used to limit the scan."
                  (group (and "_"
                              (any "A-Z" "a-z" "0-9"))
                         (zero-or-more (any "A-Z" "a-z" "0-9" "_"))
-                        (optional (or "?" "!"))))
+                        (optional (or "?" "!")))
+                 (or symbol-end "-" "+" "/"))
```

Comment:
I don't know how to explain it well (because I don't understand the fontification process enough) but I will try.

Refer to emacs wiki article on regexes: http://www.emacswiki.org/emacs/RegularExpression#toc1

The `rx` without the symbol-end compiles down to:

```lisp
"\\_<\\(_[0-9A-Za-z][0-9A-Z_a-z]*[!?]?\\)"
```

As you can see, there is no boundary/whitespace/punctuation check in the regexp. That means that partial words will match and hence will be fontified. 

Adding `symbol-end`:

```
"\\_<\\(_[0-9A-Za-z][0-9A-Z_a-z]*[!?]?\\_>\\)"
```

(From the wiki, you can see that "\\<_" corresponds to `symbol-start` and "\\_>" corresponds to `symbol-end`)

EDIT: For emacs, the `:`, `/`, `-`, `+` characters are not matched by the `symbol-end`.

Adding only the `symbol-end` will be like this:

![screenshot from 2015-12-19 00 35 54](https://cloud.githubusercontent.com/assets/4231743/11910835/a29d5658-a5e8-11e5-819b-32638f23b153.png)

Which is not right. 

Changing `symbol-end` to `(or symbol-end "-" "/" "+") yields:

![screenshot from 2015-12-19 00 35 02](https://cloud.githubusercontent.com/assets/4231743/11910843/d40b67ca-a5e8-11e5-9dc8-5afcc3dd6f39.png)